birdhouse/components/weaver/config/proxy/docker-compose-extra.yml

```diff
@@ -7,6 +7,6 @@ services:
       - ./components/weaver/config/proxy/conf.extra-service.d:/etc/nginx/conf.extra-service.d/weaver:ro
       # because of mounting path naming restrictions (see note in 'worker' definition),
       # we must add the custom path on top of named 'wps_outputs' volume of other birds for the proxy to expose results
-      - ${WEAVER_WPS_OUTPUTS_DIR}:/pavics-data/wps_outputs/weaver:ro
+      - ${WEAVER_WPS_OUTPUTS_DIR}:${THREDDS_SERVICE_DATA_LOCATION_ON_CONTAINER}/wps_outputs/weaver:ro
```

Comment:
Not sure what `THREDDS` has to do here for `weaver`/`proxy` service combination regarding WPS outputs.

Same comment for all other locations edited in [`91af29e` (#359)](https://github.com/bird-house/birdhouse-deploy/pull/359/commits/91af29eff88393f5a7d4c1778c8e5d90a8e8615d) that are not directly related to `thredds` service.

Reply:
Aren't we putting the WPS outputs under "service data" location that is then served by THREDDS as a separate catalog? (this seems to be the case now).

Reply:
Either we rename them, or we simply make the location in-docker a hardcoded `/data/wps_outputs/<bird>` for these cases. 
In the case of THREDDS, I understand that the in-docker location could matter because it allows different configurations for managing multiple datasets, but in WPS birds, there can only be 1 location for WPS outputs. It is really irrelevant where they are mounted, and it forces platform admins to learn how each bird behaves (which is not consistent). Making flexible path editable in docker for these cases can lead to more configuration errors than it seems.

Reply:
> It is really irrelevant where they are mounted, and it forces platform admins to learn how each bird behaves (which is not consistent).

But WPS outputs directory and user workspaces directory now need to be on the same mount point right? (for cowbird to synchronize WPS outputs to user workspaces) In this case, it seems to me that moving the variables out of THREDDS to another location and renaming them would be the way to go.

Reply:
Sorry, my bad. We are talking about location on container.

Reply:
Weaver needs exact mounting on both sides when running its child processes. Once the results are generated, the same WPS outputs on the server can be mounted in other services any way they want. However, we align the paths in a similar fashion to what THREDDS will do, such that the request URL looks somewhat similar. 

Cowbird and Jupyter should mount WPS-outputs and user-worspaces in a similar fashion to be resolved correctly from the point of view of the user directory while using the notebook. 

Other cases (so far), can mount however they desire as they do not interact with other data locations.

Reply:
I am still quite confused, I admit the operations of weaver are quite opaque to me. At [first](https://github.com/bird-house/birdhouse-deploy/pull/359#discussion_r1358917967) it read to me like the issue is that variables from the THREDDS component are controlling the working of the weaver component and that was a sacrilege. Now, it sounds to me that you are saying the changes are will break something. Could you please clarify which is it?

Reply:
For the case above where my comment was placed, `${WEAVER_WPS_OUTPUTS_DIR}:${THREDDS_SERVICE_DATA_LOCATION_ON_CONTAINER}/wps_outputs/weaver:ro` is applied under the `proxy` service. This will not break `weaver` service, because it only defines how to mount the WPS outputs generated by Weaver at some point, for access through the `/wpsoutputs` HTTPS alias. 

Under `weaver` service however, it must be mounted exactly as `${WEAVER_WPS_OUTPUTS_DIR}:${WEAVER_WPS_OUTPUTS_DIR}`, or it will break how it runs its processes.

In a similar fashion, I know that some birds like `finch` mount WPS outputs only as `wpsoutputs:/data/wpsoutputs` because it automatically nests its outputs under a sub-dir named `finch` (contrary to Weaver which must indicate the final `/weaver`), while others like `hummingbird` use `wps_outputs:/opt/birdhouse/var/lib/pywps/outputs` due to configuration particularities. Their respectives volumes/dirs are mounted in `proxy` such that they "align" natually as: 
```yaml
${PAVICS_FQDN_PUBLIC}/wpsoutputs
    /weaver
    /finch
    /hummingbird
    ...
```

The takeaway is that each service could have implementation-specific requirements, and trying to fit a single variable in each case is probably not feasable. In some cases, it can outright break the service. For these cases, I prefer to leave hardcoded mount locations, and remove the chances of a node admin "breaking" them unconsciously by changing configurable values. This is particularly important because "broken" paths do not necessarily raise an error, but could simply place outputs in wrong locations, and that can be hard to diagnose considering each service-specific logic/code-base.

Reply:
Thanks for the clarification. I've made some reversions, do these look good? My only remaining concern now is that I prefer not to use the term `pavics-data`, but we can think about that after we've settled that nothing is breaking with these latest changes.

Reply:
Looks good.

Reply:
Great! Done. Can you approve the PR?

Reply:
@dchandan I started to be confused by all the changes.  Can you confirm if you change all the default values to something else, the different pieces of the stack still work together?  

https://github.com/bird-house/birdhouse-deploy/blob/91af29eff88393f5a7d4c1778c8e5d90a8e8615d/birdhouse/config/thredds/default.env#L11-L18 

Meaning
* we can browse Thredds catalog from a web browser
* we can add/remove permissions to Thredds catalogs in Magpie
* a WPS service can create a wps_output that is accessible via the proxy and Thredds
* whatever other workflows that Weaver and Cowbird supports that I am not very familiar with that @fmigneault  can fill in

Reply:
To test all of this, you can use this notebook: 
https://github.com/Ouranosinc/pavics-sdi/blob/master/docs/source/notebook-components/weaver_example.ipynb

It will test Weaver, which relays a call to hummingbird WPS at the same time, using a NetCDF from the THREDDS catalog, and the resulting WPS output file should be available through the `/wpsoutput` endpoint.

Reply:
Uisng the default (unset) values for the new variables, operations seems to behave correctly.
https://github.com/bird-house/birdhouse-deploy/pull/359#issuecomment-1771124315

A test with overrides to validate everything would be good to make sure alternative setups work as intended.

Reply:
Thanks for running the tests. Should I still check WPS outputs using the notebook you linked above or is that check run as part of tests you initiated?

Reply:
> Thanks for running the tests. Should I still check WPS outputs using the notebook you linked above or is that check run as part of tests you initiated?

@dchandan FYI the automated tests use the default values.  To test that all the overrides works (which also indirectly test whether we forgot any replacement of hardcode to variable), that has to be done manually.

Reply:
@dchandan @tlvu @fmigneault 

I've run the test notebook (https://github.com/Ouranosinc/pavics-sdi/blob/master/docs/source/notebook-components/weaver_example.ipynb) with non-default values for all the variables and the notebook ran without any issues. 

The only thing I had change in the notebook was the `WEAVER_TEST_DEFAULT_FILE` variable so that it could use the updated URL (since I changed `THREDDS_SERVICE_DATA_URL_PATH` from the default `birdhouse` for the test)